src/react/useGraffy.js

```diff
@@ -29,6 +29,14 @@ const retrieveResult = async (promise, setState) => {
   }
 };
 
+/** @type {(query: {
+ *  [key in string]: [{
+ *    first?: number,
+ *    after?: string,
+ *    last?: number,
+ *    before?: string
+ *  }, Object]
+ * }, params: any) => [any, boolean, Error]} */
```

Comment:
Hmm, the type of `query` here is not really complete, unfortunately.

Graffy has four data structures: PorcelainQuery, PorcelainGraph, PlumbingQuery and PlumbingGraph. This one is a PorcelainQuery. The type of state is [PorcelainGraph, boolean, Error].

They are all recursive types.

```ts
type PorcelainQuery = 
  number | boolean |
  { [key: string]: PorcelainQuery } |
  [ RangeParams, PorcelainQuery ]

type RangeParams = { first?: number, ... }
```

```ts
type PorcelainGraph = 
  <any scalar value> |
  { [key: string]: PorcelainGraph } |
  PageData

class PageData extends Array<PorcelainGraph> {
  nextRange?: RangeParams,
  prevRange?: RangeParams
}
```
The nextRange and prevRange will be added when #19 is solved, currently there is a pageInfo.

You can see the documentation of the Plumbing data structures here: https://github.com/usegraffy/graffy/blob/master/src/common/graph/Readme.md